src/Symfony/Component/Process/Process.php

```diff
@@ -358,6 +358,11 @@ public function start(?callable $callback = null, array $env = []): void
 
         try {
             $process = @proc_open($commandline, $descriptors, $this->processPipes->pipes, $this->cwd, $envPairs, $this->options);
+
+            // Backwards compatibility. To be removed in version 8.0
+            if (!\is_resource($process) && \is_array($commandline)) {
+                $process = @proc_open($this->buildShellCommandline($commandline), $descriptors, $this->processPipes->pipes, $this->cwd, $envPairs, $this->options);
```

Comment:
we should also prefix with `exec`, either here or in buildShellCommandline I suppose

Reply:
Done in 1941795d8ab8320b61332ec447457cae5678b6a0

Windows can never reach here because `$commandline` is never an array at this point there.